gumby/modules/community_launcher.py

```diff
@@ -1,5 +1,14 @@
 from abc import ABCMeta, abstractmethod
 
+from Tribler.community.search.community import SearchCommunity
```

Comment:
I considered some of this too when I moved them. If your scenario imports any of the dispersy providers (tribler_module or dispersy_module), you will also import GumbySession and thus Session from Tribler. So the chances of needing tribler on your path are 99.9% already. (All scenario experiments to date have this requirement)

A case could be made for moving the GumbySession to only be used by the tribler_module. (it used to be that way in gumby) Some communities behave differently depending on the existance of the session. I thought this undesirable since that means that tribler_module and dispersy_module are not interchangabe. But I agree that it might not be 100% obvious that importing dispersy_module in a scenario also causes an import of Tribler. (But after moving the GumbySession to tribler_module you will also need to move all the session config objects there, since that requires Tribler on your path too. And then the community modules need separate code paths to configure stuff depending on which dispersy provider is loaded)

Regarding point 2. I hadn't considered this at all. It might also save some resources (less parsing, less memory needed) on each instance.